### ISSUE TITLE
VSS snapshot - slow file list building on windows XP 2003 during

### ISSUE THREAD

**Issue by kenkendk** [duplicati/duplicati#435]
_From [marco.de...@gmail.com](https://code.google.com/u/117277005276043801124/) on June 28, 2011 10:46:19_

Tested executed on many different installation of Windows Xp, Windows 2003, Windows 7)
## On Windows XP:   
1. select a enough populated folder (e.g. "c:\Documents and Settrings" [in my case 2,11 GB with 2.281 folders and 12.406 files])
2. setup a backup profile with "snapshot-policy" set to "required"
3. start the backup
4. after 30 minutes of "Building File List"..... and "dllhost" process at high CPU usage (the COM request to VSS provider), it finally start with the "Processing"...

I'm using the last 1.2.1 x86 version

The problem is not related to any particular backed type; so i can say that this is a source-side problem.

The performance goes slightly better on Winddows 2003 (just a little) and certainly better on Windows 7.

So i've tried to run the same profile but with then "Microsoft VSHADOW.EXE 2.2 - Volume Shadow Copy sample client" (vshadow.exe) and (dosdev.exe) with this steps:
1. at cmd prompt:
    vshadow.exe -exec="&#37;comspec&#37;" c:
2. take note of  - Shadow copy device name: \?\GLOBALROOT\Device\HarddiskVolumeShadowCopy1   (so in my case i'll use the "\?\GLOBALROOT\Device\HarddiskVolumeShadowCopy1")
3. at cmd prompt:
    dosdev B: \?\GLOBALROOT\Device\HarddiskVolumeShadowCopy1    (the drive B: in often unused nowadays but you can use another free drive-letter)
4. now we have the snapshot-volume mounted as B: (not browsable by windows explorer but available to you system [  you my try "dir b:\"  ])
5. at cmd prompt:
    "&#37;ProgramFiles&#37;\Duplicati\Duplicati.exe"
6. In dupliucati GUI, edit the profile just used before and manual substitute in the source path the C: with B:
7. Obviously also remove (in advanced settings) the "snapshot-policy" previously set to "required"
8. Launch the profile
9. I notice that (instead of 30 minutes) it stopped with "Building file list" after just 1 minute and then start with "Processing" and seems also copying more quick.
10. Quit the Duplicati.exe    (so it return to the prompt)
11. at cmd prompt:
    dosdev -r -d B:      (for remove the drive letter mapping)
    exit                 (to return to previous environment and thus deleting the snapshot)

In conclusion the COM instance (alphavss) seem slow performing if compared at VSHADOW.EXE  (take a look at http://vscsc.sourceforge.net/ ) and this is more noticeable on old (but still used) system as Windows 2003 and dramatically evident on Windows XP.

Attacched:
- vshadow.exe ["vc xp 32_64 288 vshadow.exe"](xp version 32bit - 288K)
- vshadow.exe ["vc 2003 32_64 344 vshadow.exe"](2003 version 32bit - 344K)
- dosdev.exe

**Attachment:** [vc xp 32_64 288 vshadow.exe vc 2003 32_64 344 vshadow.exe dosdev.exe](http://code.google.com/p/duplicati/issues/detail?id=435)

_Original issue: http://code.google.com/p/duplicati/issues/detail?id=435_
